kratos/python_scripts/rms_output_process.py

```diff
@@ -0,0 +1,126 @@
+
+
+    def ExecuteInitialize(self):
+        """Initialize the non historical variables and the measuring of computational time."""
+        for variable in self.nonhistorical_variables:
+            KM.VariableUtils().SetNonHistoricalVariableToZero(variable, self.model_part.Nodes)
+        self.start_time = time.time()
+
+
+    def IsOutputStep(self):
+        """Check if the current time step is near enough to the specified printing times."""
+        time = self.model_part.ProcessInfo[KM.TIME]
+        return time >= self.next_output
+
+
+    def PrintOutput(self):
+        """Write the values into the file."""
+        self._WriteAverageError()
+        self.next_output += self.time_frequency
+
+
+    def Check(self):
+        """Check the correctness of the input parameters."""
+        for variable in self.variables + self.nonhistorical_variables:
+            if not isinstance(variable, KM.DoubleVariable):
+                raise Exception("This process is expecting only double or component variables")
+
+
+    def _InitializeOutputFile(self):
+        file_path = Path(self.settings["filename"].GetString()).with_suffix('.dat')
+        file_path.touch(exist_ok=True)
+        header = self._GetFileHeader()
+        if file_path.stat().st_size == 0:
+            with open(file_path, 'w') as file:
+                file.write(header)
+        else:
+            existing_header = ''
+            with open(file_path, 'r') as file:
+                for _ in range(2):
+                    existing_header += file.readline()
+            if existing_header != header:
+                msg = self.__class__.__name__ + ": "
+                msg += "The specified fields mismatch\n"
+                msg += "Existing header:\n"
+                msg += existing_header
+                msg += "Specified header:\n"
+                msg += header
+                raise Exception(msg)
+
+
+    def _GetFileHeader(self):
+        header = "# RMS for model part '{}'\n".format(self.model_part.Name)
+        header += "label datetime num_nodes num_elems elem_size time_step time computational_time"
+        for variable in self.variables + self.nonhistorical_variables:
+            header += ' ' + variable.Name()
+        header += '\n'
+        return header
+
+
+    def _WriteAverageError(self):
+        data  = self.settings["analysis_label"].GetString() + ' '
+        data += str(time.time()) + ' '
+        data += str(self.model_part.NumberOfNodes()) + ' '
+        data += str(self.model_part.NumberOfElements()) + ' '
+        data += str(self.model_part.Elements.__iter__().__next__().GetGeometry().Length()) + ' '
+        data += str(self.model_part.ProcessInfo[KM.DELTA_TIME]) + ' '
+        data += str(self.model_part.ProcessInfo[KM.TIME]) + ' '
+        data += str(time.time() - self.start_time)
+
+        for variable in self.variables:
+            value = ST.SpatialMethods.Historical.NormMethods.RootMeanSquare(self.model_part, variable)
```

Comment:
```suggestion
            value = ST.SpatialMethods.Historical.NormMethods.RootMeanSquare(self.model_part, variable)
```

I think this will fail because, there needs to be another argument describing the type of the norm.

You can refer the following for an example.
https://kratosmultiphysics.github.io/Kratos/pages/Applications/Statistics_Application/Domains/Spatial/spatial_methods.html

Reply:
I've not tried, this is a draft based on a custom process in shallow app ([here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/ShallowWaterApplication/python_scripts/postprocess/convergence_output_process.py))